lib/src/list.dart

```diff
@@ -83,6 +83,17 @@ class ManagedRealmList<T extends Object?> with RealmEntity, ListMixin<T> impleme
     }
   }
 
+  @override
+  bool remove(covariant T element) {
+    if (element is RealmObject && !element.isManaged) {
+      throw RealmStateError('Cannot call remove on a managed list with an element that is an unmanaged object');
+    }
+    final index = indexOf(element);
+    final found = index > 0;
+    if (found) removeAt(index);
```

Comment:
Purely stylistic, but I hate those one-line ifs 😭

Reply:
dartfmt rules .. get over it

Reply:
You can always put `{` around to make it look nice 😜